### PR TITLE
[bitnami/magento] add OpenSearch as a search engine option

### LINKED ISSUE #63907: support opensearch as magento search engine option
### Name and Version

bitnami/magento:2.4.6

### What is the problem this feature will solve?

provision Magento with OpenSearch specific setup options

### What is the feature you are proposing to solve the problem?

I have a PR to support it but I am not allowed to push it
```console
$ git push origin --set-upstream magento-add-opensearch-as-a-search-engine-option
ERROR: Permission to bitnami/containers.git denied to vtkstef.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

### What alternatives have you considered?

elasticsearch but its licensing change makes it a problematic choice

### LINKED ISSUE #63916: bitnami/elasticsearch:8.12.2 arm64 image failed to run
### Name and Version

bitnami/elasticsearch:8.12.2

### What architecture are you using?

arm64

### What steps will reproduce the bug?

1. `docker pull bitnami/elasticsearch:8.12.2`
2. `docker run -itd bitnami/elasticsearch:8.12.2`
3. container exited with error message: "could not find java in ES_JAVA_HOME at /opt/bitnami/java/bin/java"

### What do you see instead?
```
elasticsearch 09:56:16.70 INFO  ==> 
elasticsearch 09:56:16.71 INFO  ==> Welcome to the Bitnami elasticsearch container
elasticsearch 09:56:16.71 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
elasticsearch 09:56:16.71 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
elasticsearch 09:56:16.71 INFO  ==> 
elasticsearch 09:56:16.72 INFO  ==> ** Starting Elasticsearch setup **
elasticsearch 09:56:16.75 WARN  ==> Setting ELASTICSEARCH_IS_DEDICATED_NODE is disabled.
elasticsearch 09:56:16.75 WARN  ==> ELASTICSEARCH_NODE_ROLES will be ignored and Elasticsearch will asume all different roles.
elasticsearch 09:56:16.76 INFO  ==> Configuring/Initializing Elasticsearch...
elasticsearch 09:56:16.77 INFO  ==> Setting default configuration
elasticsearch 09:56:16.80 INFO  ==> Configuring Elasticsearch cluster settings...
could not find java in ES_JAVA_HOME at /opt/bitnami/java/bin/java
```

### PR DESCRIPTION
**Pull request by vtkstef**
### Description of the change

add OpenSearch as a search engine option for Magento. Option `ealasticsearch5 ` and `elasticsearch6` are
no longer supported during setup/bootstrap in magento 2.4.6

### Benefits

OpenSearch has a more permissive open source license (hence the fork from elastic search by amazon) 

### Possible drawbacks

n/a

### Applicable issues

- fixes #63907 

### Additional information

Option `ealasticsearch5 ` and `elasticsearch6` areno longer supported during setup/bootstrap in magento 2.4.6

### CODE REVIEW

**Comment by vtkstef** [comment]
@carrodher FYI can you please reset the timer on this PR. Thanks

**Comment by mdhont** [comment]
We are going to review this logic internally and we will notify you in this PR when there is any news.
Thank you very much for the contribution!

**Comment by vtkstef** [comment]
Ping @mdhont

File: bitnami/magento/2/debian-12/rootfs/opt/bitnami/scripts/libmagento.sh [hunk 1]
```diff
@@ -204,6 +216,31 @@ magento_initialize() {
...
         fi
+        if [[ "$MAGENTO_SEARCH_ENGINE" =~ ^opensearch ]]; then
+            oss_host="$MAGENTO_OPENSEARCH_HOST"
+            oss_port="$MAGENTO_OPENSEARCH_PORT_NUMBER"
+            oss_user="$MAGENTO_OPENSEARCH_USER"
+            oss_pass="$MAGENTO_OPENSEARCH_PASSWORD"
+            # Define whether Opensearch auth is enabled
+            local oss_auth="0"
+            is_boolean_yes "$MAGENTO_OPENSEARCH_ENABLE_AUTH" && oss_auth="1"
+            # Elasticsearch configuration is stored in the database, so we only need to specify for 'setup:install'
```
Comment (dgomezleon) [inline bitnami/magento/2/debian-12/rootfs/opt/bitnami/scripts/libmagento.sh:227]:
```suggestion
            # OpenSearch configuration is stored in the database, so we only need to specify for 'setup:install'
```

File: bitnami/magento/2/debian-12/rootfs/opt/bitnami/scripts/libmagento.sh [hunk 2]
```diff
@@ -228,6 +265,10 @@ magento_initialize() {
             info "Trying to connect to Elasticsearch"
             magento_wait_for_es_connection "$es_host" "$es_port"
         fi
+        if [[ "$MAGENTO_SEARCH_ENGINE" =~ ^opensearch ]]; then
+            info "Trying to connect to Opensearch"
+            magento_wait_for_es_connection "$oss_host" "$oss_port"
```
Comment (dgomezleon) [inline bitnami/magento/2/debian-12/rootfs/opt/bitnami/scripts/libmagento.sh:270]: I would rename this function to something generic. E.g. `magento_wait_for_search_engine_connection` or similar.

File: bitnami/magento/2/debian-12/rootfs/opt/bitnami/scripts/libmagento.sh [hunk 3]
```diff
@@ -357,6 +398,13 @@ magento_initialize() {
             magento_wait_for_es_connection "$es_host" "$es_port"
         fi
 
+        if [[ "$MAGENTO_SEARCH_ENGINE" =~ ^opensearch ]]; then
+            oss_host="$MAGENTO_OPENSEARCH_HOST"
+            oss_port="$MAGENTO_OPENSEARCH_PORT_NUMBER"
+            info "Trying to connect to Elasticsearch"
```
Comment (dgomezleon) [inline bitnami/magento/2/debian-12/rootfs/opt/bitnami/scripts/libmagento.sh:404]:
```suggestion
            info "Trying to connect to OpenSearch"
```

**Review by dgomezleon — CHANGES_REQUESTED** [review]
Thanks a lot for this contribution. I just added a couple of minor suggestions.

**Pull request closed without merging**